conans/client/tools/oss.py

```diff
@@ -24,12 +25,38 @@ def args_to_string(args):
 
 def cpu_count(output=None):
     output = default_output(output, 'conans.client.tools.oss.cpu_count')
+
+    def is_running_in_docker():
+        try:
+            cgroup = "/proc/self/cgroup"
+            return (os.path.exists('/.dockerenv') or
+                    (os.path.isfile(cgroup) and any('docker' in line for line in open(cgroup))))
+        except:
+            return False
+
+    def docker_cpu_count():
+        base_cpus = "/sys/fs/cgroup/cpu"
+        try:
+            with open(os.path.join(base_cpus, "cpu.cfs_quota_us"), 'r') as quota_fd:
+                cfs_quota_us = int(quota_fd.read())
+            with open(os.path.join(base_cpus, "cpu.cfs_period_us"), 'r') as period_fd:
+                cfs_period_us = int(period_fd.read())
+            if cfs_quota_us > -1 and cfs_period_us > 0:
+                return math.ceil(cfs_quota_us / cfs_period_us)
```

Comment:
if `cfs_quota_us` is zero, we may end up with `-j0`, which will have an error: ` option requires a positive integral argument`

Reply:
also, result of `math.ceil` is double (like `1.0`), which wouldn't be a valid target for `make -j`

Reply:
good catch, but users are able to set `--cpus=0.5`, which will be promoted to `1` by `math.ceil`. Also, if someone sets cpus to `0` it's a prone error IMO.

Reply:
no, I mean `ceil(0.5) = 1.0`, not just `1`, so it will be `-j1.0` which is invalid

Reply:
Are you sure?

```
Python 3.7.2 (default, Mar 27 2019, 09:22:51) 
[GCC 8.2.1 20181127] on linux
import math
>>> math.ceil(0.5)
1
>>> math.ceil(0.3)
1
>>> math.ceil(0.4)
1
>>> math.ceil(0.1)
1
>>> math.ceil(0)
0
>>> math.ceil(1)
1
>>> math.ceil(1.0)
1
```

Reply:
yes:
```
python -c "import math; print(math.ceil(1))"
1.0
python -c "import math; print(math.ceil(1.5))"
2.0
```

Reply:
also from [doc](https://docs.python.org/2/library/math.html):
```
math.ceil(x)
Return the ceiling of x as a float, the smallest integer value greater than or equal to x.
```

Reply:
now I see, you are running python 2 ... well let's truncate. Thanks for finding it.